### ISSUE TITLE
RDP to Windows on custom port

### ISSUE THREAD

**Issue by anobar** [FreeRDP/Remmina#1509]
Dear developers,

Once again I have to bother because RDP connection via custom port does not work. Similar to  #237. Connection over 3389 standard worked. Change to port 33389 on both sides does not work.

Remmina version 1.2.0-rcgit-27 (git rcgit-27)
Xubuntu 16.04.4 LTS
FreeRDP version 2.0.0-dev2 (n/a) (installed freerdp-x11 after first appearance of problem)
Connecting to Windows 10 1709 16299.248

Console output of Remmina:
Gtk-Message: GtkDialog mapped without a transient parent. This is discouraged.
[19:10:51:633] [20273:23105] [INFO][com.freerdp.client.common.cmdline] - loading channelEx rdpdr
[19:10:51:633] [20273:23105] [INFO][com.freerdp.client.common.cmdline] - loading channelEx rdpsnd
[19:10:51:633] [20273:23105] [INFO][com.freerdp.client.common.cmdline] - loading channelEx cliprdr
[19:10:51:634] [20273:23105] [INFO][com.freerdp.client.common.cmdline] - loading channelEx drdynvc
[19:11:06:649] [20273:23105] [ERROR][com.freerdp.core] - failed to connect to 192.168.0.123
[19:11:06:649] [20273:23105] [ERROR][com.freerdp.core.nego] - Protocol Security Negotiation Failure
[19:11:06:649] [20273:23105] [ERROR][com.freerdp.core] - freerdp_set_last_error ERRCONNECT_SECURITY_NEGO_CONNECT_FAILED [0x0002000C]
[19:11:06:649] [20273:23105] [ERROR][com.freerdp.core.connection] - Error: protocol security negotiation or connection failure
Gtk-Message: GtkDialog mapped without a transient parent. This is discouraged.

Error message: Unable to establish a connection to RDP server beeblebrox.

rm .config/freerdp/knownhosts did not help.

Thanks and kind regards,
Axel

**Comment by arniotis** [comment]
Same problem, except that I'm using a standard port (3389). Problem started right after installing windows updates on windows 10 machine this morning. I can connect without problem using remmina 1.1.2 from xubuntu 17.10, but I can't connect using 1.2.0-rcgit-27. Connections to Windows 7 and 2012R2 servers work without problem. The error I get is:

<pre>
[18:49:00:363] [11349:11387] [INFO][com.freerdp.client.common.cmdline] - loading channelEx rdpdr
[18:49:00:363] [11349:11387] [INFO][com.freerdp.client.common.cmdline] - loading channelEx rdpsnd
[18:49:00:363] [11349:11387] [INFO][com.freerdp.client.common.cmdline] - loading channelEx cliprdr
[18:49:00:363] [11349:11387] [INFO][com.freerdp.client.common.cmdline] - loading channelEx drdynvc
[18:49:00:378] [11349:11387] [ERROR][com.freerdp.core] - freerdp_set_last_error ERRCONNECT_PASSWORD_CERTAINLY_EXPIRED [0x0002000F]
[18:49:00:378] [11349:11387] [ERROR][com.freerdp.core.transport] - BIO_read returned an error: error:14094438:SSL routines:ssl3_read_bytes:tlsv1 alert internal error
0002000F 00000003
</pre>

My guess is that it is a freerdp error; connecting to the same server from the command line with freerdp 2.0.0-dev2  gets me a bogus <pre>[19:07:00:023] [11754:11755] [ERROR][com.freerdp.core] - freerdp_set_last_error ERRCONNECT_PASSWORD_CERTAINLY_EXPIRED [0x0002000F]
[19:07:00:023] [11754:11755] [ERROR][com.freerdp.core.transport] - BIO_read returned an error: error:14094438:SSL routines:ssl3_read_bytes:tlsv1 alert internal error
[19:07:00:024] [11754:11755] [ERROR][com.freerdp.client.x11] - Freerdp connect error exit status 1
*** Error in `xfreerdp': double free or corruption (out): 0x00007f6dcc036a80 ***</pre> 

message since the windows updates. I have no problem connecting if I downgrade to freerdp 1.1.0-beta1 . I'll take my report to FreeRDP, but I'm adding this here as well in case it helps Axel.

Michael

**Comment by antenore** [comment]
@arniotis see #1513

**Comment by giox069** [comment]
@anobar  ERRCONNECT_SECURITY_NEGO_CONNECT_FAILED usually appears when there is no answer from the other side.
Try to sniff traffic with **tcpdump**: install tcpdump package, open another shell window and type
```sudo tcpdump -nn host _the_remote_server_ip_address_``` (put the remote server IP address here, without the port). Go to the remmina window and connect to the remote server. You should see some tcpdump lines like these:
```
17:24:34.994651 IP 192.168.56.3.59224 > XX.XX.XX.XX:port: Flags [S], seq 873003252, win 29200, options [mss 1460,sackOK,TS val 4058086829 ecr 0,nop,wscale 10], length 0
```
Close tcpdump with CTRL+C.
Copy the tcpdump output and paste here.

**Comment by arniotis** [comment]
@antenore - Indeed, I'm up and running again. Thanks!

**Comment by anobar** [comment]
At the moment it even does not work on the standard port. Updated the system (Ubuntu 16.04), hope no other bugs with Win10 03-2018 update interfere. The requested output is below.

Thanks, axel

 xfreerdp --version
This is FreeRDP version 2.0.0-dev2 (n/a)
[user@computer: ~ ]
 sudo tcpdump -nn -vv host 192.168.179.104
tcpdump: listening on enp0s25, link-type EN10MB (Ethernet), capture size 262144 bytes
Gtk-Message: GtkDialog mapped without a transient parent. This is discouraged.
[16:44:29:585] [9993:27774] [INFO][com.freerdp.client.common.cmdline] - loading channelEx cliprdr
[16:44:29:585] [9993:27774] [INFO][com.freerdp.client.common.cmdline] - loading channelEx drdynvc
16:44:29.879318 IP (tos 0x0, ttl 64, id 41713, offset 0, flags [DF], proto TCP (6), length 60)
    192.168.179.108.51962 > 192.168.179.104.3389: Flags [S], cksum 0xe854 (incorrect -> 0x7ff6), seq 905181651, win 29200, options [mss 1460,sackOK,TS val 179395362 ecr 0,nop,wscale 8], length 0
16:44:29.958419 IP (tos 0x0, ttl 128, id 28702, offset 0, flags [DF], proto TCP (6), length 52)
    192.168.179.104.3389 > 192.168.179.108.51962: Flags [S.], cksum 0x5c22 (correct), seq 207821916, ack 905181652, win 64000, options [mss 1460,nop,wscale 0,nop,nop,sackOK], length 0
16:44:29.958470 IP (tos 0x0, ttl 64, id 41714, offset 0, flags [DF], proto TCP (6), length 40)
    192.168.179.108.51962 > 192.168.179.104.3389: Flags [.], cksum 0xe840 (incorrect -> 0x967b), seq 1, ack 1, win 115, length 0
16:44:29.958537 IP (tos 0x0, ttl 64, id 41715, offset 0, flags [DF], proto TCP (6), length 59)
    192.168.179.108.51962 > 192.168.179.104.3389: Flags [P.], cksum 0xe853 (incorrect -> 0x8461), seq 1:20, ack 1, win 115, length 19
16:44:29.995458 IP (tos 0x0, ttl 128, id 28703, offset 0, flags [DF], proto TCP (6), length 59)
    192.168.179.104.3389 > 192.168.179.108.51962: Flags [P.], cksum 0x59af (correct), seq 1:20, ack 20, win 63981, length 19
16:44:29.995574 IP (tos 0x0, ttl 64, id 41716, offset 0, flags [DF], proto TCP (6), length 40)
    192.168.179.108.51962 > 192.168.179.104.3389: Flags [.], cksum 0xe840 (incorrect -> 0x9655), seq 20, ack 20, win 115, length 0
16:44:29.995915 IP (tos 0x0, ttl 64, id 41717, offset 0, flags [DF], proto TCP (6), length 364)
    192.168.179.108.51962 > 192.168.179.104.3389: Flags [P.], cksum 0xe984 (incorrect -> 0xc554), seq 20:344, ack 20, win 115, length 324
16:44:30.028280 IP (tos 0x0, ttl 127, id 28704, offset 0, flags [DF], proto TCP (6), length 1205)
    192.168.179.104.3389 > 192.168.179.108.51962: Flags [P.], cksum 0x8739 (correct), seq 20:1185, ack 344, win 63657, length 1165
16:44:30.029017 IP (tos 0x0, ttl 64, id 41718, offset 0, flags [DF], proto TCP (6), length 166)
    192.168.179.108.51962 > 192.168.179.104.3389: Flags [P.], cksum 0xe8be (incorrect -> 0x95ee), seq 344:470, ack 1185, win 126, length 126
16:44:30.034698 IP (tos 0x0, ttl 127, id 28705, offset 0, flags [DF], proto TCP (6), length 91)
    192.168.179.104.3389 > 192.168.179.108.51962: Flags [P.], cksum 0xc4c3 (correct), seq 1185:1236, ack 470, win 63531, length 51
16:44:30.077796 IP (tos 0x0, ttl 64, id 41719, offset 0, flags [DF], proto TCP (6), length 40)
    192.168.179.108.51962 > 192.168.179.104.3389: Flags [.], cksum 0xe840 (incorrect -> 0x8fc8), seq 470, ack 1236, win 126, length 0
16:44:33.927830 IP (tos 0x0, ttl 64, id 41720, offset 0, flags [DF], proto TCP (6), length 126)
    192.168.179.108.51962 > 192.168.179.104.3389: Flags [P.], cksum 0xe896 (incorrect -> 0xdc6e), seq 470:556, ack 1236, win 126, length 86
16:44:33.953650 IP (tos 0x0, ttl 127, id 28707, offset 0, flags [DF], proto TCP (6), length 280)
    192.168.179.104.3389 > 192.168.179.108.51962: Flags [P.], cksum 0x6564 (correct), seq 1236:1476, ack 556, win 63445, length 240
16:44:33.953725 IP (tos 0x0, ttl 64, id 41721, offset 0, flags [DF], proto TCP (6), length 40)
    192.168.179.108.51962 > 192.168.179.104.3389: Flags [.], cksum 0xe840 (incorrect -> 0x8e79), seq 556, ack 1476, win 135, length 0
16:44:33.954450 IP (tos 0x0, ttl 64, id 41722, offset 0, flags [DF], proto TCP (6), length 774)
    192.168.179.108.51962 > 192.168.179.104.3389: Flags [P.], cksum 0xeb1e (incorrect -> 0xe46b), seq 556:1290, ack 1476, win 135, length 734
16:44:33.958460 IP (tos 0x0, ttl 127, id 28708, offset 0, flags [DF], proto TCP (6), length 71)
    192.168.179.104.3389 > 192.168.179.108.51962: Flags [P.], cksum 0xc744 (correct), seq 1476:1507, ack 1290, win 62711, length 31
[16:44:33:666] [9993:27774] [ERROR][com.freerdp.core] - freerdp_set_last_error ERRCONNECT_PASSWORD_CERTAINLY_EXPIRED [0x0002000F]
[16:44:33:666] [9993:27774] [ERROR][com.freerdp.core.transport] - BIO_read returned an error: error:14094438:SSL routines:ssl3_read_bytes:tlsv1 alert internal error
16:44:33.958676 IP (tos 0x0, ttl 64, id 41723, offset 0, flags [DF], proto TCP (6), length 71)
    192.168.179.108.51962 > 192.168.179.104.3389: Flags [P.], cksum 0xe85f (incorrect -> 0x13b9), seq 1290:1321, ack 1507, win 135, length 31
16:44:33.958769 IP (tos 0x0, ttl 64, id 41724, offset 0, flags [DF], proto TCP (6), length 40)
    192.168.179.108.51962 > 192.168.179.104.3389: Flags [F.], cksum 0xe840 (incorrect -> 0x8b5c), seq 1321, ack 1507, win 135, length 0
16:44:33.962096 IP (tos 0x0, ttl 127, id 28709, offset 0, flags [DF], proto TCP (6), length 40)
    192.168.179.104.3389 > 192.168.179.108.51962: Flags [.], cksum 0x970a (correct), seq 1507, ack 1322, win 62680, length 0
Gtk-Message: GtkDialog mapped without a transient parent. This is discouraged.
16:44:41.542042 IP (tos 0x0, ttl 128, id 28710, offset 0, flags [DF], proto TCP (6), length 40)
    192.168.179.104.3389 > 192.168.179.108.51918: Flags [R.], cksum 0x9e08 (correct), seq 815646656, ack 2769007660, win 0, length 0

**Comment by giox069** [comment]
@anobar: ERRCONNECT_PASSWORD_CERTAINLY_EXPIRED is the error caused by 13 March Windows uptates, see issues #1512 and #1513. So your freerdp is not updated. Can you retry to update the system with the usual `apt-get update && apt-get dist-upgrade`, reboot and retry connect on the standard 3389 port?
It you still receive ERRCONNECT_PASSWORD_CERTAINLY_EXPIRED on port 3389 check the version of your freerdp2 packages with `apt-cache policy libfreerdp-plugins-standard` and `apt-cache policy libfreerdp1` and post result in issue #1513.
If you are able to connect to port 3389, then proceed with the tcpdump for the connection in the different port.

**Comment by anobar** [comment]
OK, standard port 3389 works now, certificate accepted.
Change on windows machine 33893, restart.
In remmina port added :33893 to address.
Connect from remmina, takes quite while without response. Then message "Unable to connect to RDP server <computer name>. TCP dump below.

Thanks and hope it helps.


$ sudo tcpdump -nn host -vv 192.168.179.104
tcpdump: listening on wlp3s0, link-type EN10MB (Ethernet), capture size 262144 bytes
Gtk-Message: GtkDialog mapped without a transient parent. This is discouraged.
loading channel cliprdr
19:08:24.054523 IP (tos 0x0, ttl 64, id 19217, offset 0, flags [DF], proto TCP (6), length 60)
    192.168.179.111.43094 > 192.168.179.104.33893: Flags [S], cksum 0x8487 (correct), seq 2881560245, win 29200, options [mss 1460,sackOK,TS val 3529809831 ecr 0,nop,wscale 7], length 0
19:08:25.058820 IP (tos 0x0, ttl 64, id 19218, offset 0, flags [DF], proto TCP (6), length 60)
    192.168.179.111.43094 > 192.168.179.104.33893: Flags [S], cksum 0x809b (correct), seq 2881560245, win 29200, options [mss 1460,sackOK,TS val 3529810835 ecr 0,nop,wscale 7], length 0
19:08:27.074819 IP (tos 0x0, ttl 64, id 19219, offset 0, flags [DF], proto TCP (6), length 60)
    192.168.179.111.43094 > 192.168.179.104.33893: Flags [S], cksum 0x78bb (correct), seq 2881560245, win 29200, options [mss 1460,sackOK,TS val 3529812851 ecr 0,nop,wscale 7], length 0
19:08:29.282793 ARP, Ethernet (len 6), IPv4 (len 4), Request who-has 192.168.179.104 tell 192.168.179.111, length 28
19:08:29.348454 ARP, Ethernet (len 6), IPv4 (len 4), Reply 192.168.179.104 is-at 98:5f:d3:3a:ad:01, length 28
19:08:31.330823 IP (tos 0x0, ttl 64, id 19220, offset 0, flags [DF], proto TCP (6), length 60)
    192.168.179.111.43094 > 192.168.179.104.33893: Flags [S], cksum 0x681b (correct), seq 2881560245, win 29200, options [mss 1460,sackOK,TS val 3529817107 ecr 0,nop,wscale 7], length 0
19:08:39.522826 IP (tos 0x0, ttl 64, id 19221, offset 0, flags [DF], proto TCP (6), length 60)
    192.168.179.111.43094 > 192.168.179.104.33893: Flags [S], cksum 0x481b (correct), seq 2881560245, win 29200, options [mss 1460,sackOK,TS val 3529825299 ecr 0,nop,wscale 7], length 0
19:08:55.650827 IP (tos 0x0, ttl 64, id 19222, offset 0, flags [DF], proto TCP (6), length 60)
    192.168.179.111.43094 > 192.168.179.104.33893: Flags [S], cksum 0x091b (correct), seq 2881560245, win 29200, options [mss 1460,sackOK,TS val 3529841427 ecr 0,nop,wscale 7], length 0
19:09:03.029785 IP (tos 0x0, ttl 1, id 32251, offset 0, flags [none], proto IGMP (2), length 40, options (RA))
    192.168.179.104 > 224.0.0.22: igmp v3 report, 1 group record(s) [gaddr 239.255.255.250 is_ex { }]
19:09:06.101721 IP (tos 0x0, ttl 1, id 32252, offset 0, flags [none], proto IGMP (2), length 40, options (RA))
    192.168.179.104 > 224.0.0.22: igmp v3 report, 1 group record(s) [gaddr 224.0.0.251 is_ex { }]
19:09:12.040895 IP (tos 0x0, ttl 1, id 32253, offset 0, flags [none], proto IGMP (2), length 40, options (RA))
    192.168.179.104 > 224.0.0.22: igmp v3 report, 1 group record(s) [gaddr 224.0.0.252 is_ex { }]
19:09:27.906830 IP (tos 0x0, ttl 64, id 19223, offset 0, flags [DF], proto TCP (6), length 60)
    192.168.179.111.43094 > 192.168.179.104.33893: Flags [S], cksum 0x8b1a (correct), seq 2881560245, win 29200, options [mss 1460,sackOK,TS val 3529873683 ecr 0,nop,wscale 7], length 0
19:09:33.026794 ARP, Ethernet (len 6), IPv4 (len 4), Request who-has 192.168.179.104 tell 192.168.179.111, length 28
19:09:33.036300 ARP, Ethernet (len 6), IPv4 (len 4), Reply 192.168.179.104 is-at 98:5f:d3:3a:ad:01, length 28
unable to connect to <computer name>:33893
Error: protocol security negotiation or connection failure
Gtk-Message: GtkDialog mapped without a transient parent. This is discouraged.


Versions:
Win10 with 2018-03 update and version 16299.309
xubuntu 17.10

$ apt-cache policy libfreerdp-plugins-standard
libfreerdp-plugins-standard:
  Installed: 1.1.0~git20140921.1.440916e+dfsg1-15ubuntu1
  Candidate: 1.1.0~git20140921.1.440916e+dfsg1-15ubuntu1
  Version table:
 *** 1.1.0~git20140921.1.440916e+dfsg1-15ubuntu1 500
        500 http://de.archive.ubuntu.com/ubuntu artful/main amd64 Packages
        100 /var/lib/dpkg/status

**Comment by giox069** [comment]
There is a firewall blocking traffic: your client is sending TCP SYN packets to 192.168.179.104 on port 33893, but there are no answers on the opposite direction. Try to temporarily disable the windows 10 firewall and try to reconnect. Let us know if with firewall disabled remmina can connect.

**Comment by anobar** [comment]
Yes, it was the firewall! Somehow makes sense looking back. For changing the port I found a how to on the microsoft support pages that shows which entry to change in the registry. The standard RDP firewall rules are for port 3389 and do not respect the registry change. 

Finally I added a rule the following way:
Start > PC Settings > Network & Internet > Windows Firewall > Advanced Settings
Left menu select 'Inbound Rules'
Right menu 'Actions' > New Rule...
Create a rule of type 'Port'
Select 'TCP' and enter in 'Specific local ports:' the port number (33893 in my case)
'Allow the connection'
In my case 'Private' is enough
Give it a name like 'Remote Desktop - User Mode (my modification)' and corresponding description.
Finish

All the best
